cartofacade/carto_facade.go

```diff
@@ -0,0 +1,176 @@
+// Package cartofacade contains the api to call into CGO
+package cartofacade
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	cgoApi "github.com/viamrobotics/viam-cartographer/cartofacade/internal/capi"
+)
+
+// WorkType defines the grpc call that is being made.
+type WorkType int64
+
+const (
+	// Initialize can be used to represent the viam_carto_init call in c.
+	Initialize WorkType = iota
+	// Start can be used to represent the viam_carto_start call in c.
+	Start
+	// Stop can be used to represent the viam_carto_stop call in c.
+	Stop
+	// Terminate can be used to represent the viam_carto_terminate in c.
+	Terminate
+	// AddSensorReading can be used to represent the viam_carto_add_sensor_reading in c.
+	AddSensorReading
+	// Position can be used to represent the viam_carto_get_position call in c.
+	Position
+	// InternalState can be used to represent the viam_carto_get_internal_state call in c.
+	InternalState
+	// PointCloudMap can be used to represent the viam_carto_get_point_cloud_map in c.
+	PointCloudMap
+)
+
+// InputType defines the type being provided as input to the work.
+type InputType int64
+
+const (
+	// Sensor represents a sensor name input into c funcs.
+	Sensor InputType = iota
+	// Reading represents a lidar reading input into c funcs.
+	Reading
+	// Timestamp represents the timestamp input into c funcs.
+	Timestamp
+)
+
+// Response defines the result of one piece of work that can be put on the result channel.
+type Response struct {
+	result interface{}
+	err    error
+}
+
+// CartoFacade represents a queue to consume work from and enforce one call into C at a time.
+type CartoFacade struct {
+	WorkChannel     chan WorkItem
+	CartoLib        *cgoApi.CartoLibInterface
```

Comment:
Why is this a pointer to an interface rather than an interface itself? Interface pointers are an antipattern in go which I would really like to avoid.

could we make it so that `CartoFacade.CartoLib` takes the interface value itself, not a pointer to an interface value:

More on why: https://glucn.medium.com/golang-an-interface-holding-a-nil-value-is-not-nil-bb151f472cc7

Reply:
fixed! thanks for the additional info